### PR TITLE
Use custom user identity per Hadoop server (non-secure)

### PR DESCRIPTION
**Pull request by frankgh**
Allow users to set the `pxf.service.user.name` property to define
the hadoop login user on a per server basis. By default, hadoop
will use the OS user name.

PXF provides a template `$PXF_CONF/templates/pxf-site.xml`.
To configure the `pxf.service.user.name` property follow these
steps:

1. Copy the pxf-site.xml template to your non-secure hadoop
server configuration

        cp $PXF_CONF/templates/pxf-site.xml $PXF_CONF/servers/non-secure-hadoop-server

2. Edit $PXF_CONF/servers/non-secure-hadoop-server-configuration/pxf-site.xml
by replacing the value for the `pxf.service.user.name` property
to your hadoop user.

### CODE REVIEW

File: concourse/scripts/pxf_common.bash [hunk 1]
```diff
@@ -4,6 +4,8 @@ GPHOME=/usr/local/greenplum-db-devel
 PXF_HOME=${GPHOME}/pxf
 MDD_VALUE=/data/gpdata/master/gpseg-1
 PXF_COMMON_SRC_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+PROXY_USER=${PROXY_USER:-pxfuser}
```
Comment (denalex) [inline concourse/scripts/pxf_common.bash]: are you sure we want to default to `pxfuser` and not `gpadmin` ? If people turn off impersonation in PXF, and if PXF is run as `gpadmin`, then they would need to have `gpadmin` as the proxy user in hadoop, which I think should be default for our tests

File: concourse/scripts/pxf_common.bash [hunk 2]
```diff
@@ -200,11 +202,11 @@ function setup_impersonation() {
 		echo 'Impersonation is enabled, adding support for gpadmin proxy user'
 		cat > proxy-config.xml <<-EOF
 			<property>
-			  <name>hadoop.proxyuser.gpadmin.hosts</name>
+			  <name>hadoop.proxyuser.pxfuser.hosts</name>
```
Comment (denalex) [inline concourse/scripts/pxf_common.bash]: should this be using the variable ?
```suggestion
			  <name>hadoop.proxyuser.${PROXY_USER}.hosts</name>
```

File: concourse/scripts/pxf_common.bash [hunk 3]
```diff
@@ -200,11 +202,11 @@ function setup_impersonation() {
...
 			  <value>*</value>
 			</property>
 			<property>
-			  <name>hadoop.proxyuser.gpadmin.groups</name>
+			  <name>hadoop.proxyuser.pxfuser.groups</name>
```
Comment (denalex) [inline concourse/scripts/pxf_common.bash]: same here

File: server/pxf-service/src/templates/user/templates/pxf-site.xml [hunk 4]
```diff
@@ -12,7 +12,7 @@
     </property>
     <property>
         <name>pxf.service.user.name</name>
-        <value>${user}</value>
+        <value>${user.name}</value>
```
Comment (denalex) [inline server/pxf-service/src/templates/user/templates/pxf-site.xml]: right ! how was it working till this change, though ?

Comment (frankgh) [reply server/pxf-service/src/templates/user/templates/pxf-site.xml]: we haven't used it for non-kerberos until now. I had to fix it for this work

File: concourse/scripts/pxf_common.bash (hunk 1 above)
Comment (frankgh) [reply concourse/scripts/pxf_common.bash]: hmm, how is our job without impersonation working then?
https://ud.ci.gpdb.pivotal.io/teams/main/pipelines/dev:pxf_fguerrero-custom_username_per_server_5X_STABLE/jobs/test_pxf_no_impersonation/builds/5

Comment (frankgh) [reply concourse/scripts/pxf_common.bash]: Ok, I looked at this. If impersonation is disabled, then we don't need to configure a proxy user.

**Comment by shivzone** [comment]
What is the motivation behind having a separate pxf user per server ?

**Comment by frankgh** [comment]
@shivzone the proxy user configured in hadoop might not necessarily be called `gpadmin`. We had   requests for allowing configuring a different proxy user other than gpadmin ([for example](https://groups.google.com/a/greenplum.org/d/msg/gpdb-users/u7QdDEt18QE/54RlOFHjAgAJ)).

With multi hadoop support, we support any proxy user configured in Hadoop for each different server. It's possible that each Hadoop cluster will have a different proxy user configured for PXF.

**Comment by shivzone** [comment]
I initially though of proxy user for pxf as one specific proxy for the entire pxf service. But i see your point. If there is a new hadoop server with an already configured proxyuser say `proxy1` it would be useful to allow them to use that for the given server

File: concourse/scripts/pxf_common.bash (hunk 3 above)
Comment (frankgh) [reply concourse/scripts/pxf_common.bash]: done

File: concourse/scripts/pxf_common.bash (hunk 2 above)
Comment (frankgh) [reply concourse/scripts/pxf_common.bash]: done

**Pull request merged**